### ISSUE TITLE
Loss Weights in repository different from those in your paper

### ISSUE THREAD

**Issue by humannumber196** [sicxu/Deep3DFaceRecon_pytorch#41]
Dear Authors,

Thanks for sharing your code! I found in this repository the w_feat is set to 1.0, w_id=0.2, and w_tex=1.7e-2 while in your paper they are set to w_per=0.2, w_alpha=1.0 and w_gamma=1.7e-3 (this w_gamma is the w_tex in this repository). Could you please tell me why they are different from what's in your paper? Besides I found neither the settings in your paper nor the settings in this repository can give me a model as good as what you released. Could you please help me to find what's wrong? I used MTCNN to extract the 5 facial landmarks and followed your data preparation procedure to prepare the training data. For the images in 300W-LP and LS3D-W datasets, I loaded the ground truth 68 landmarks directly and used the method extract_5p() in this repository to extract the 5 facial landmarks. I didn't balance the pose and race distributions, will this influence much? Thank you very much for reading this issue!

**Comment by sicxu** [comment]
You may adjust the loss weight for a quick check. https://github.com/sicxu/Deep3DFaceRecon_pytorch/commit/73d491102af6731bded9ae6b3cc7466c3b2e9e48#r55395513
https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/091e84551a6f78cc5046320b5229fde2e5e902c1/models/facerecon_model.py#L50-L60
A large feature/perceptual loss may lead to bad performance. We tried to increase the w_id, the reconstruction error increases on the MICC Florence dataset.

**Comment by humannumber196** [comment]
> You may adjust the loss weight for a quick check. [73d4911#r55395513](https://github.com/sicxu/Deep3DFaceRecon_pytorch/commit/73d491102af6731bded9ae6b3cc7466c3b2e9e48#r55395513)
> 
> https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/091e84551a6f78cc5046320b5229fde2e5e902c1/models/facerecon_model.py#L50-L60
> 
> 
> A large feature/perceptual loss may lead to bad performance. We tried to increase the w_feat, the reconstruction error increased on the MICC Florence dataset.

OK, thanks for your reply!

**Comment by LY-Donkey-Burger** [comment]
> > You may adjust the loss weight for a quick check. [73d4911#r55395513](https://github.com/sicxu/Deep3DFaceRecon_pytorch/commit/73d491102af6731bded9ae6b3cc7466c3b2e9e48#r55395513)
> > https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/091e84551a6f78cc5046320b5229fde2e5e902c1/models/facerecon_model.py#L50-L60
> > 
> > A large feature/perceptual loss may lead to bad performance. We tried to increase the w_feat, the reconstruction error increased on the MICC Florence dataset.
> 
> OK, thanks for your reply!

Hi，could you get a model as good as the one published by the authors

**Comment by LY-Donkey-Burger** [comment]
> > You may adjust the loss weight for a quick check. [73d4911#r55395513](https://github.com/sicxu/Deep3DFaceRecon_pytorch/commit/73d491102af6731bded9ae6b3cc7466c3b2e9e48#r55395513)
> > https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/091e84551a6f78cc5046320b5229fde2e5e902c1/models/facerecon_model.py#L50-L60
> > 
> > A large feature/perceptual loss may lead to bad performance. We tried to increase the w_feat, the reconstruction error increased on the MICC Florence dataset.
> 
> OK, thanks for your reply!

Hi ，could you get a model as good as the one published by the author ？

**Comment by humannumber196** [comment]
> > > You may adjust the loss weight for a quick check. [73d4911#r55395513](https://github.com/sicxu/Deep3DFaceRecon_pytorch/commit/73d491102af6731bded9ae6b3cc7466c3b2e9e48#r55395513)
> > > https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/091e84551a6f78cc5046320b5229fde2e5e902c1/models/facerecon_model.py#L50-L60
> > > 
> > > A large feature/perceptual loss may lead to bad performance. We tried to increase the w_feat, the reconstruction error increased on the MICC Florence dataset.
> > 
> > 
> > OK, thanks for your reply!
> 
> Hi ，could you get a model as good as the one published by the author ？

I can't...

**Comment by LY-Donkey-Burger** [comment]
> > > > You may adjust the loss weight for a quick check. [73d4911#r55395513](https://github.com/sicxu/Deep3DFaceRecon_pytorch/commit/73d491102af6731bded9ae6b3cc7466c3b2e9e48#r55395513)
> > > > https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/091e84551a6f78cc5046320b5229fde2e5e902c1/models/facerecon_model.py#L50-L60
> > > > 
> > > > A large feature/perceptual loss may lead to bad performance. We tried to increase the w_feat, the reconstruction error increased on the MICC Florence dataset.
> > > 
> > > 
> > > OK, thanks for your reply!
> > 
> > 
> > Hi ，could you get a model as good as the one published by the author ？
> 
> I can't...

How much the reconstructed error of the model you get is different from the author's

**Comment by LY-Donkey-Burger** [comment]
> 

How much the reconstructed error of the model you get is different from the author's？

**Comment by LY-Donkey-Burger** [comment]
> > > > You may adjust the loss weight for a quick check. [73d4911#r55395513](https://github.com/sicxu/Deep3DFaceRecon_pytorch/commit/73d491102af6731bded9ae6b3cc7466c3b2e9e48#r55395513)
> > > > https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/091e84551a6f78cc5046320b5229fde2e5e902c1/models/facerecon_model.py#L50-L60
> > > > 
> > > > A large feature/perceptual loss may lead to bad performance. We tried to increase the w_feat, the reconstruction error increased on the MICC Florence dataset.
> > > 
> > > 
> > > OK, thanks for your reply!
> > 
> > 
> > Hi ，could you get a model as good as the one published by the author ？
> 
> I can't...

How much the reconstructed error of the model you get is different from the author's
